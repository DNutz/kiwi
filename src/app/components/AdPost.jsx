import React, { Component } from 'react';
import { Card, CardActions, CardHeader, CardMedia, CardText } from 'material-ui/Card';

export class AdPost extends Component {
    /**
     * Component constructor
     * @param  {object} props is an object properties of component
     */
    constructor(props) {
        super(props);
    }

    render() {
        const { image } = this.props;
        return (
            <div style={{ backgroundColor: '#fff', border: '1px solid #dddfe2', borderRadius: '7px' }}>
                <CardMedia>
                    <img src={image} />
                </CardMedia>
            </div>
        );
    }
}

export default AdPost