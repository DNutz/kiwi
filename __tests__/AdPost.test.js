import React from "react";
import expect from "expect";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AdPost from "../src/app/components/AdPost";
import { shallow } from "enzyme"; 
import renderer from "react-test-renderer";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";


Enzyme.configure({ adapter: new Adapter() });

describe("Component: AdPost", () => {
    it("renders without crashing", () => {
        const wrapper = shallow(<AdPost />);
        expect(wrapper.length).toEqual(1);
    });    

    it("renders correctly", () => {
      const component = renderer.create(
          <MuiThemeProvider>
              <AdPost image="src/public/images/postAds/yourAdHere.jpg"/>
          </MuiThemeProvider>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("receives a proper string in image", () => {
        const wrapper = Enzyme.mount(<MuiThemeProvider><AdPost image="src/public/images/postAds/yourAdHere.jpg" /></MuiThemeProvider>);
        expect(
            typeof(wrapper.childAt(0).prop("image")) === "string"
        ).toEqual(true);
    });

    it("receives the correct string in image", () => {
        const wrapper = Enzyme.mount(<MuiThemeProvider><AdPost image="src/public/images/postAds/yourAdHere.jpg" /></MuiThemeProvider>);
        expect(
            String(wrapper.childAt(0).prop("image")) === "src/public/images/postAds/yourAdHere.jpg"
        ).toEqual(true);
    });

    it("has correct default state type for image", () => {
        const wrapper = Enzyme.mount(<MuiThemeProvider><AdPost/></MuiThemeProvider>);
        expect(
            typeof(wrapper.childAt(0).prop("image")) === "undefined"
        ).toEqual(true);
    });

    it("has correct default state for image", () => {
        const wrapper = Enzyme.mount(<MuiThemeProvider><AdPost/></MuiThemeProvider>);
        expect(
            String(wrapper.childAt(0).prop("image")) === "undefined"
        ).toEqual(true);
    });
});

