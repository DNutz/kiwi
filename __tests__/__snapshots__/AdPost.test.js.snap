// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AdPost renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#fff",
      "border": "1px solid #dddfe2",
      "borderRadius": "7px",
    }
  }
>
  <div
    style={
      Object {
        "muiPrepared": true,
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "muiPrepared": true,
        }
      }
    >
      <img
        src="src/public/images/postAds/yourAdHere.jpg"
        style={
          Object {
            "maxWidth": "100%",
            "minWidth": "100%",
            "muiPrepared": true,
            "verticalAlign": "top",
            "width": "100%",
          }
        }
      />
    </div>
    
  </div>
</div>
`;
